vertexai/multimodal-all/multimodalall.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// multimodalall shows an example of understanding as multimodal input a video having audio
+package multimodalall
+
+// [START generativeaionvertexai_gemini_all_modalities]
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"path/filepath"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+// multimodalPrompt is a sample prompt type consisting of one video, one image, and a text question.
+type multimodalPrompt struct {
+	// video and image are Google Cloud Storage paths starting with "gs://"
+	video, image string
+	// question is the question asked to the model
+	question string
+}
+
+// generateMultimodalContent generates a response into w, based upon the multimodal prompt
+// provided.
+func generateMultimodalContent(w io.Writer, prompt multimodalPrompt, projectID, location, modelName string) error {
```

Comment:
I see the function name `generateMultimodalContent` is the same as generativeaionvertexai_gemini_video_with_audio from #4110.

question: Why do both samples exist? Would it make more sense to associate the same sample code with both region tags?
suggestion: If two samples are needed, rename one for clarity in the event someone is comparing them side-by-side.

This may benefit from feedback from @gericdong

Reply:
We need language parity for the doc pages, usually Python comes first and then other languages are added in tabs at the same place: https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/send-multimodal-prompts#all_modalities

Yes I will rename the func, for clarity/disambiguation